server/grpc_service.go

```diff
@@ -1900,59 +1901,66 @@ func (s *GrpcServer) StoreGlobalConfig(_ context.Context, request *pdpb.StoreGlo
 	if !res.Succeeded {
 		return &pdpb.StoreGlobalConfigResponse{Error: &pdpb.Error{Type: pdpb.ErrorType_UNKNOWN, Message: "failed to execute StoreGlobalConfig transaction"}}, errors.Errorf("failed to execute StoreGlobalConfig transaction")
 	}
-	return &pdpb.StoreGlobalConfigResponse{}, err
+	return &pdpb.StoreGlobalConfigResponse{}, nil
 }
 
 // LoadGlobalConfig load global config from etcd
 func (s *GrpcServer) LoadGlobalConfig(ctx context.Context, request *pdpb.LoadGlobalConfigRequest) (*pdpb.LoadGlobalConfigResponse, error) {
-	// TODO: complete this function with new implementation
-	return &pdpb.LoadGlobalConfigResponse{}, nil
+	configPath := request.GetConfigPath()
+	r, err := s.client.Get(ctx, s.GetEtcdPath(configPath), clientv3.WithPrefix())
+	if err != nil {
+		item := &pdpb.GlobalConfigItem{Name: configPath, Error: &pdpb.Error{Type: pdpb.ErrorType_UNKNOWN, Message: err.Error()}}
+		return &pdpb.LoadGlobalConfigResponse{Items: []*pdpb.GlobalConfigItem{item}}, err
+	} else if len(r.Kvs) == 0 {
+		msg := "key " + configPath + " not found"
+		item := &pdpb.GlobalConfigItem{Name: configPath, Error: &pdpb.Error{Type: pdpb.ErrorType_GLOBAL_CONFIG_NOT_FOUND, Message: msg}}
```

Comment:
Why return this error if the config does not exist? It can happen in the normal logic.

Reply:
You are right, we don't need to worry about loading null and just put all the error in func's error to return.